packages/hdwallet-native/src/adapter.ts

```diff
@@ -1,49 +1,44 @@
 import * as core from "@shapeshiftoss/hdwallet-core";
-import { create } from "./native";
+import { create, NativeHDWallet } from "./native";
 
-type NativeAdapterArgs = {
-  mnemonic: string;
+export type NativeAdapterArgs = {
+  mnemonic?: string;
   deviceId: string;
 };
 
 export class NativeAdapter {
   keyring: core.Keyring;
   deviceId: string;
-
-  private mnemonic: string;
+  #mnemonic: string;
 
   private constructor(keyring: core.Keyring, args: NativeAdapterArgs) {
     this.keyring = keyring;
-    this.mnemonic = args.mnemonic;
     this.deviceId = args.deviceId;
+    this.#mnemonic = args.mnemonic;
   }
 
   static useKeyring(keyring: core.Keyring, args: NativeAdapterArgs) {
     return new NativeAdapter(keyring, args);
   }
 
   async initialize(): Promise<number> {
-    let wallet = this.keyring.get(this.deviceId);
+    let wallet = this.keyring.get<NativeHDWallet>(this.deviceId);
```

Comment:
If we generic to extend `HDWallet` we should probably update the rest of the wallets as well? A side question would be is this necessary?

Reply:
The reason this is here is because the `events` property is unique to NativeHDWallet. If I leave it as a generic HDWallet, there's a compilation error for accessing a property that doesn't exist. `keyring.get` does have a type constraint that all generic types must extend HDWallet.